op-batcher/batcher/channel_manager_test.go

```diff
@@ -726,5 +730,272 @@ func TestChannelManager_TxData_ForcePublish(t *testing.T) {
 	// The channel should be full and ready to send
 	require.Len(t, m.channelQueue, 1)
 	require.True(t, m.channelQueue[0].IsFull())
+}
+
+func newBlock(parent *types.Block, numTransactions int) *types.Block {
+	var rng *rand.Rand
+	if parent == nil {
+		rng = rand.New(rand.NewSource(123))
+	} else {
+		rng = rand.New(rand.NewSource(int64(parent.Header().Number.Uint64())))
+	}
+	block := derivetest.RandomL2BlockWithChainId(rng, numTransactions, defaultTestRollupConfig.L2ChainID)
+	header := block.Header()
+	if parent == nil {
+		header.Number = new(big.Int)
+		header.ParentHash = common.Hash{}
+		header.Time = 1675
+	} else {
+		header.Number = big.NewInt(0).Add(parent.Header().Number, big.NewInt(1))
+		header.ParentHash = parent.Header().Hash()
+		header.Time = parent.Header().Time + 2
+	}
+	return types.NewBlock(header, block.Body(), nil, trie.NewStackTrie(nil), types.DefaultBlockConfig)
+}
+
+func newChain(numBlocks int) []*types.Block {
+	blocks := make([]*types.Block, numBlocks)
+	blocks[0] = newBlock(nil, 10)
+	for i := 1; i < numBlocks; i++ {
+		blocks[i] = newBlock(blocks[i-1], 10)
+	}
+	return blocks
+}
+
+// TestChannelManagerUnsafeBytes tests the unsafe bytes in the channel manager
+// by adding blocks to the unsafe block queue, adding them to a channel,
+// and then sealing the channel. It asserts on the final state of the channel
+// manager and tracks the unsafe DA estimate as blocks move through the pipeline.
+func TestChannelManagerUnsafeBytes(t *testing.T) {
+
+	type testCase struct {
+		blocks                        []*types.Block
+		batchType                     uint
+		compressor                    string
+		afterAddingToUnsafeBlockQueue int64
+		afterAddingToChannel          int64
+		afterSealingChannel           int64
+	}
+
+	a := newBlock(nil, 3)
+	b := newBlock(a, 3)
+	c := newBlock(b, 3)
+
+	emptyA := newBlock(nil, 0)
+	emptyB := newBlock(emptyA, 0)
+	emptyC := newBlock(emptyB, 0)
+
+	twentyBlocks := newChain(20)
+	tenBlocks := newChain(10)
+
+	testChannelManagerUnsafeBytes := func(t *testing.T, tc testCase) {
+		cfg := ChannelConfig{
+			MaxFrameSize:    120000 - 1,
+			TargetNumFrames: 5,
+			BatchType:       tc.batchType,
+			CompressorConfig: compressor.Config{
+				TargetOutputSize: 120000,
+				CompressionAlgo:  derive.Brotli10,
+			},
+		}
+
```

Comment:
I think you can leave out setting the `CompressorConfig` because you later call `InitShadowCompressor` or `InitRatioCompressor` which overrides both fields anyways.
```suggestion
		cfg := ChannelConfig{
			MaxFrameSize:    120000 - 1,
			TargetNumFrames: 5,
			BatchType:       tc.batchType,
		}
```

Reply:
[cedca41](https://github.com/ethereum-optimism/optimism/pull/16950/commits/cedca4164c900973db391ac39c1aae62a0c3168a) we still need it for singular batches